### PR TITLE
Add documentation on input file parsing

### PR DESCRIPTION
**Pull request by andrew-platt**
** Summary **

This is a documentation only PR to add documentation on how input files are parsed.

Also add missing documentation on the IfW input file change from v2.4.0 to v2.5.0.  This was overlooked during PR #596.

The input file parsing section is viewable here: https://ap-openfast.readthedocs.io/en/b-inputfileformatdocs/source/user/input_file_overview.html

The change to the API-change table is viewable here: https://ap-openfast.readthedocs.io/en/b-inputfileformatdocs/source/user/api_change.html#modified-in-openfast-v2-4-0

### CODE REVIEW

**Comment by ebranlard** [comment]
It's looking good, here are some potential updates to consider: 
- it seems the input file by module table is incomplete, no? AeroDyn and Inflowind are using Key+Value
- should we say something about lists? Suggesting that they should be comma-separated (even if fortran doesn't care)

**Review by ebranlard — APPROVED** [review]

**Comment by andrew-platt** [comment]
@ebranlard 

- yes, the input file by module table is incomplete.  I started listing all the input files and quickly realized that it was going to take a long time to do.  So I just left it there, but commented out so it won't render on the readthedocs.  @rafmudaf suggested just listing the ones using the Key+Value input format.
- Excellent point about lists.  I can add more detail on that.  Also maybe some of the common issues with the `outlist` section with the new format:
   - put quotes around names, otherwise make sure any trailing description text has a comment character before it starts

File: docs/source/user/api_change.rst [hunk 1]
```diff
@@ -135,10 +135,22 @@ InflowWind     7    VFlowAng                  0   VFlowAng    - Upflow angle (de
 
    -  The input file parser is updated to a keyword/value pair based input.
       Each entry must have a corresponding keyword with the same spelling as
-      expected
+      expected. See :numref:`input_file_overview` for an overview.
    -  Driver code includes ability to convert between wind types
 
 
+Modified in OpenFAST v2.4.0
```
[the remark this answers is not in the document]
Comment (andrew-platt) [reply docs/source/user/api_change.rst:142]: I think one should be modified (as in pre-existing fields) and the other should be added.  I'll check on this.

[the remark this answers is not in the document]
Comment (andrew-platt) [reply docs/source/user/api_change.rst:142]: Good catch.  That was a mistake.  Fixed.

File: docs/source/user/input_file_overview.rst [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+would be read as `2` and the `20` and everything after it ignored.
+
+This format and associated parsing methodology is somewhat limited in informing
+the user of errors in parsing, and limited the ability to pass entire inpute
+files as text strings from another code (such as a Python driver code).
+
+.. _sec_format_key_value:
+
+Key + Value input files
```
[the remark this answers is not in the document]
Comment (andrew-platt) [reply docs/source/user/input_file_overview.rst:34]: Added a table with these.

**Comment by rdamiani** [comment]
HI Andy, is there any quick way to revert it back to "column value" in a local branch of OFAST by chance? Probably not, but I thought I'd check.
I was perhaps the largest user of that format we developed with Marshall, and I still use it a lot,
thank you all for your hard work.

**Pull request merged**